pkg/consuldp/consul_dataplane.go

```diff
@@ -152,6 +156,11 @@ func (cdp *ConsulDataplane) Run(ctx context.Context) error {
 		return fmt.Errorf("failed to run proxy: %w", err)
 	}
 
+	if err = cdp.startDNSProxy(ctx); err != nil {
+		cdp.logger.Error("failed to start the dns proxy", "error", err)
+		return err
+	}
+
```

Comment:
would be a good idea to call a `defer dnsServer.Stop()` so if Run returns ( due to proxy exiting or the xds server exiting), dns server is gracefully stopped.

Reply:
I've moved this ahead of the envoy start.